pkg/hintrunner/zero/zerohint_math_test.go

```diff
@@ -479,5 +481,94 @@ func TestZeroHintMath(t *testing.T) {
+					{Name: "value", Kind: apRelative, Value: feltInt64(3042)},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newAssert250bitsHint(ctx.operanders["low"], ctx.operanders["high"], ctx.operanders["value"])
+				},
+				check: allVarValueEquals(map[string]*fp.Element{
+					"low":  feltInt64(3042),
+					"high": feltInt64(0),
+				}),
+			},
+
+			{
+				operanders: []*hintOperander{
+					{Name: "low", Kind: reference, Value: addrBuiltin(starknet.RangeCheck, 0)},
+					{Name: "high", Kind: reference, Value: addrBuiltin(starknet.RangeCheck, 1)},
+					{Name: "value", Kind: fpRelative, Value: feltInt64(4938538853994)},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newAssert250bitsHint(ctx.operanders["low"], ctx.operanders["high"], ctx.operanders["value"])
+				},
+				check: allVarValueEquals(map[string]*fp.Element{
+					"low":  feltInt64(4938538853994),
+					"high": feltInt64(0),
+				}),
+			},
+
+			{
+				operanders: []*hintOperander{
+					{Name: "low", Kind: reference, Value: addrBuiltin(starknet.RangeCheck, 0)},
+					{Name: "high", Kind: reference, Value: addrBuiltin(starknet.RangeCheck, 1)},
+					{Name: "value", Kind: apRelative, Value: feltString("348329493943842849393993999999231222222222")},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newAssert250bitsHint(ctx.operanders["low"], ctx.operanders["high"], ctx.operanders["value"])
+				},
+				check: allVarValueEquals(map[string]*fp.Element{
+					"low":  feltString("220632583722801270961776596532341902734"),
+					"high": feltInt64(1023),
+				}),
+			},
+
+			{
+				operanders: []*hintOperander{
+					{Name: "low", Kind: reference, Value: addrBuiltin(starknet.RangeCheck, 0)},
+					{Name: "high", Kind: reference, Value: addrBuiltin(starknet.RangeCheck, 1)},
+					{Name: "value", Kind: apRelative, Value: feltString("348329493943842849393124453993999999231222222222")},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newAssert250bitsHint(ctx.operanders["low"], ctx.operanders["high"], ctx.operanders["value"])
+				},
+				check: allVarValueEquals(map[string]*fp.Element{
+					"low":  feltString("302658603189151847763334509038790380942"),
+					"high": feltInt64(1023648380),
+				}),
+			},
+
+			{
+				operanders: []*hintOperander{
+					{Name: "low", Kind: reference, Value: addrBuiltin(starknet.RangeCheck, 0)},
+					{Name: "high", Kind: reference, Value: addrBuiltin(starknet.RangeCheck, 1)},
+					{Name: "value", Kind: apRelative, Value: feltString("348329493943842849393124453993999999231222222222")},
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newAssert250bitsHint(ctx.operanders["low"], ctx.operanders["high"], ctx.operanders["value"])
+				},
+				check: allVarValueEquals(map[string]*fp.Element{
+					"low":  feltString("302658603189151847763334509038790380942"),
+					"high": feltInt64(1023648380),
+				}),
+			},
```

Comment:
This test case is the same as the one right above right?

Reply:
I will add a small sanity check that would protect us from such mistakes in the future.